financial_calculations/coupon_rates.py

```diff
@@ -0,0 +1,65 @@
+import numpy as np
+import pandas as pd
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from utils import (
+    get_ZCB_vector
+)
+from financial_calculations.bond_cash_flows import (
+    PMTS_PER_YEAR
+)
+
+def calculate_coupon_rate(start_date, maturity_years, par_value, forward_curve):
+    """
+    Calculate the coupon rate required to produce a par price for a bond.
+    
+    Parameters:
+    -----------
+    start_date : datetime or datetime64[D]
+        The start date of the bond.
+    maturity_years : int
+        The maturity years of the bond.
+    par_value : float
+        The par value of the bond.
+    forward_curve : tuple
+        A tuple containing two elements:
+        - spot_rate_dates (np.ndarray): Array of spot rate dates.
```

Comment:
this is not a great name.  these aren't spot rates.  the only that is actually a spot rate is the first one.  everything else is some kind of forward rate.  disc_rates and disc_dates may be better.  or something similar